lib/src/source_visitor.dart

```diff
@@ -2021,8 +2021,10 @@ class SourceVisitor extends ThrowingAstVisitor {
     _visitDirectiveMetadata(node);
     _simpleStatement(node, () {
       token(node.libraryKeyword);
-      space();
-      visit(node.name2);
+      if (node.name2 != null) {
+        space();
+        visit(node.name2);
+      }
```

Comment:
This is OK, but a simpler way to do it is:

```dart
visit(node.name2, before: space);
```

Thanks for doing this!

Reply:
Thanks! Done here: https://github.com/dart-lang/dart_style/pull/1145